build: todo-library

AnastasiiaShirova opened the pull request:
добавляется подключение бэка в todo-api-service, изменяются методы в todo-service для получения, добавления, удаления и редактирования задач, добавляется interceptor для генерации todo.id

AnastasiiaShirova committed:
build: todo-library

добавляется подключение бэка в todo-api-service, изменяются методы в todo-service для получения, добавления, удаления и редактирования задач, добавляется interceptor для генерации todo.id

AnastasiiaShirova committed:
build: todo-library

в todo-service изменяются методы для завершения/активации всех задач и для удаления выполненных задач

tamazyanarsen commented on projects/todo-library/src/lib/todo/services/todo.service.ts:
```diff
@@ -1,43 +1,53 @@
+      this.apiService.deleteTodoFromBack(todo.id).subscribe(() => {
+        currentTodoList.splice(index, 1);
+        this.todoList$.next(currentTodoList);
+      });
```
в сервисе не должно быть подписок, мы ведь обсуждали это :)
возвращай поток, например

```suggestion
      this.apiService.deleteTodoFromBack(todo.id).pipe(
        tap(() => {
          currentTodoList.splice(index, 1);
          this.todoList$.next(currentTodoList);
        })
      );
```

AnastasiiaShirova committed:
build: todo-library

добавляются storage-service и storage-interceptor для взаимодействия через localStorage

tamazyanarsen commented on projects/todo-library/src/lib/todo/services/todo.service.ts:
```diff
@@ -1,43 +1,53 @@
   );
 
+  fetchTodos(): void {
+    this.apiService.getTodos().subscribe((todos) => this.todoList$.next(todos));
```
убираем все подписки из сервисов

AnastasiiaShirova committed:
fix: todo-library

переносятся подписки из сервисов
